perf/impl/https/v0.1/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/binary"
+	"encoding/json"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/big"
+	"net"
+	"net/http"
+	"time"
+)
+
+const (
+	BlockSize = 64 << 10
+)
+
+func handleRequest(w http.ResponseWriter, r *http.Request) {
+	// Read the big-endian bytesToSend value
+	var bytesToSend uint64
+	if err := binary.Read(r.Body, binary.BigEndian, &bytesToSend); err != nil {
+		http.Error(w, "failed to read uint64 value", http.StatusBadRequest)
+		return
+	}
+
+	// Read and discard the remaining bytes in the request
+	io.Copy(io.Discard, r.Body)
+
+	buf := make([]byte, BlockSize)
+
+	for bytesToSend > 0 {
+		toSend := buf
+		if bytesToSend < BlockSize {
+			toSend = buf[:bytesToSend]
+		}
+
+		n, err := w.Write(toSend)
+		if err != nil {
+			http.Error(w, "Failed write", http.StatusInternalServerError)
+			return
+		}
+		bytesToSend -= uint64(n)
+	}
+}
+
+func runClient(serverAddr string, uploadBytes, downloadBytes uint64) (time.Duration, time.Duration, error) {
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	reqBody := make([]byte, 8+uploadBytes)
```

Comment:
? You're allocating all the bytes at once here? We could be uploading gigs. Please make a new reader that will read until `uploadBytes` have been read.

Reply:
Good catch! Fixed with c0c34f4. Note that I am using a custom `io.Reader`. Let me know in case there is a more idiomatic way using Go's http stack.